db/init.sql

```diff
@@ -4,13 +4,154 @@ CREATE DATABASE mm_db;
 CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
 
 CREATE TYPE user_role AS ENUM ('admin', 'teacher', 'student');
+CREATE TYPE attempt_state AS ENUM ('submitted', 'graded');
+-- NOTE(nrydanov): Need to think of other types together
+CREATE TYPE block_type AS ENUM ('task', 'markdown');
+
+CREATE TABLE unit_type (
+    id uuid PRIMARY KEY,
+    name varchar(64)
+);
 
 CREATE TABLE users (
-    id uuid NOT NULL PRIMARY KEY,
+    id uuid PRIMARY KEY,
+    -- Auth data
+    username varchar(32) UNIQUE NOT NULL,
     email varchar(128) UNIQUE NOT NULL,
-    name varchar(32) NOT NULL,
-    surname varchar(32) NOT NULL,
-    patronymic varchar(32),
-    role user_role NOT NULL,
-    password bytea NOT NULL
+    password bytea NOT NULL,
+    -- Identity data
+    name varchar(64) NOT NULL,
+    surname varchar(64) NOT NULL,
```

Comment:
What about optional `patronymic` field? It is ok for students to not specify their patronymic, but teachers often want to have it in their profile. It is also useful for students with poor memory

Reply:
We decided not to include this because many students may or may not have patronymic.

Reply:
I would like to hear arguments against storing a `patronymic`, but ok, it's fine then